src/FsAutoComplete.Core/SymbolCache.fs

```diff
@@ -9,6 +9,88 @@ open System.Net
 open System.IO
 open Newtonsoft.Json
 
+[<CLIMutable>]
+type SymbolUseRange = {
+    FileName: string
+    StartLine: int
+    StartColumn: int
+    EndLine: int
+    EndColumn: int
+    IsFromDefinition: bool
+    IsFromAttribute : bool
+    IsFromComputationExpression : bool
+    IsFromDispatchSlotImplementation : bool
+    IsFromPattern : bool
+    IsFromType : bool
+    SymbolFullName: string
+    SymbolDisplayName: string
+    SymbolIsLocal: bool
+}
+
+
+module PersistenCacheImpl =
+    open Microsoft.Data.Sqlite
+    open Dapper
+    open System.Data
+
+    let mutable connection : SqliteConnection option = None
+
+    let insert (connection: SqliteConnection) file (sugs: SymbolUseRange[]) =
+        if connection.State <> ConnectionState.Open then connection.Open()
+        use tx = connection.BeginTransaction()
+        let delCmd = sprintf "DELETE FROM Symbols WHERE FileName=\"%s\"" file
+        let inserCmd =
+            sprintf "INSERT INTO SYMBOLS(FileName, StartLine, StartColumn, EndLine, EndColumn, IsFromDefinition, IsFromAttribute, IsFromComputationExpression, IsFromDispatchSlotImplementation, IsFromPattern, IsFromType, SymbolFullName, SymbolDisplayName, SymbolIsLocal) VALUES
+            (@FileName, @StartLine, @StartColumn, @EndLine, @EndColumn, @IsFromDefinition, @IsFromAttribute, @IsFromComputationExpression, @IsFromDispatchSlotImplementation, @IsFromPattern, @IsFromType, @SymbolFullName, @SymbolDisplayName, @SymbolIsLocal)"
+        connection.Execute(delCmd, transaction = tx) |> ignore
+        connection.Execute(inserCmd, sugs, transaction = tx) |> ignore
+        tx.Commit()
+
+    let loadAll (connection: SqliteConnection) =
+        if connection.State <> ConnectionState.Open then connection.Open()
+        let q = "SELECT * FROM SYMBOLS"
+        let res = connection.Query<SymbolUseRange>(q)
+        res
+
+    let loadFile (connection: SqliteConnection) file =
+        if connection.State <> ConnectionState.Open then connection.Open()
+        let q = sprintf "SELECT * FROM SYMBOLS WHERE FileName=\"%s\"" file
+        let res = connection.Query<SymbolUseRange>(q)
+        res
+
+    let initializeCache dir =
+        let connectionString = sprintf "Data Source=%s/.ionide/symbolCache.db" dir
```

Comment:
if we use `/.ionide/obj` , that will be git ignored by default by pratically all `.gitignores`, so less noise

Reply:
Another option is that it's passed as argument to FSAC.

So editors can put that:
- or in their temp directory
- or can be in a dir inside %TEMP% generated for the workspace so valid until restart and automatically cleaned up.